bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/network/MagentaTVHttp.java

```diff
@@ -0,0 +1,164 @@
+            httpHeader.setProperty(HEADER_ACCEPT, "*/*");
+            response = HttpUtil.executeUrl(HttpMethod.GET, url, httpHeader, null, null, NETWORK_TIMEOUT_MS);
+            logger.trace("GET {} - Response={}", url, response);
+            return response;
+        } catch (IOException e) {
+            throw new MagentaTVException(e, "HTTP GET {0} failed: {1}", url, response);
+        }
+    }
+
+    /**
+     * Given a URL and a set parameters, send a HTTP POST request to the URL
+     * location created by the URL and parameters.
+     *
+     * @param url The URL to send a POST request to.
+     * @param urlParameters List of parameters to use in the URL for the POST
+     *            request. Null if no parameters.
+     * @param soapAction Header attribute for SOAP ACTION: xxx
+     * @param connection Header attribut for CONNECTION: xxx
+     * @return String contents of the response for the POST request.
+     * @throws MagentaTVException
+     */
+    public String httpPOST(String host, String url, String postData, String soapAction, String connection)
+            throws MagentaTVException {
+        String httpResponse = "";
+        try {
+            Properties httpHeader = new Properties();
+            httpHeader.setProperty(HEADER_CONTENT_TYPE, CONTENT_TYPE_XML);
+            httpHeader.setProperty(HEADER_ACCEPT, "");
+            httpHeader.setProperty(HEADER_USER_AGENT, USER_AGENT);
+            httpHeader.setProperty(HEADER_HOST, host);
+            if (!soapAction.isEmpty()) {
+                httpHeader.setProperty(HEADER_SOAPACTION, soapAction);
+            }
+            if (!connection.isEmpty()) {
+                httpHeader.setProperty(HEADER_CONNECTION, connection);
+            }
+
+            logger.trace("POST {} - SoapAction={}, Data = {}", url, postData, soapAction);
+            InputStream dataStream = new ByteArrayInputStream(
+                    postData.getBytes(Charset.forName(StandardCharsets.UTF_8.name())));
+            httpResponse = HttpUtil.executeUrl(HttpMethod.POST, url, httpHeader, dataStream, null, NETWORK_TIMEOUT_MS);
+            logger.trace("POST {} - Response = {}", url, httpResponse);
+            return httpResponse;
+        } catch (IOException e) {
+            throw new MagentaTVException(e, "HTTP POST {0} failed, response={1}", url, httpResponse);
+        }
+    }
+
+    /**
+     * Send raw TCP data (SUBSCRIBE command)
+     *
+     * @param remoteIp receiver's IP
+     * @param remotePort destination port
+     * @param data data to send
+     * @return received response
+     * @throws IOException
+     */
+    public String sendData(String remoteIp, String remotePort, String data) throws MagentaTVException {
+        String response = "";
+        Socket socket = null;
+        String errorMessage = "";
+
+        try {
+            // logger.trace("Sending data to {}:{}: {}", remoteIp, remotePort, data);
+            socket = new Socket();
```

Comment:
Please use a try-with-resources for the socket so that it is closed automatically

Reply:
changed